posix/stubs.c

```diff
@@ -24,17 +24,50 @@
 #include <netinet/in.h>
 #include <stdarg.h>
 #include <grp.h>
+#include <stdio.h>
+
+
+static struct {
+	char __current_locale;
+	char buffposix[6];
+	char buffc[2];
```

Comment:
I suugest using one buffer here and doing sprintf(buf, C, POSIX) on each setlocal(*, NULL) call. User program can damage content of this buffer.